### DISCUSSION TITLE
postprocessing with ESM-Tools

### DISCUSSION THREAD

**Discussion by mandresm** [esm-tools/esm_tools#755] [ESM-Tools support] [answered]
@xshi-awi wrote in an email:

>Hi Miguel
>
>I wrote a post-process script and would like to call it after each model year.
>
>Is there any possible way to add this into the runscript?
>
>The script calculate monthly mean and daily mean values of the raw 6-hourly output and afterwards delete the original file to save space.
>
>Thanks very much.

**Comment by mandresm** [comment, marked as answer]
Hi @xshi-awi,

You can checkout the explanation on how that works in our workshop: https://github.com/esm-tools/workshops/blob/main/202207_AWI_ESM-Tools/docs/pdf/7.5_workflow_manager_offline_coupler.pdf
Please, check the last slide, as it contains very valuable advices on how not to break ESM-Tools while using postprocessing.

The workshop also contains an exercise on how to implment a postprocessing scrip with the solution: https://github.com/esm-tools/workshops/blob/main/202207_AWI_ESM-Tools/exercises/7.5_workflow_manager.md

The most important part, is that you'll need to pass absolute path variables to your script so that your script can find thos files. For that you'll need this logic as specified in the exercise:
```
                      script: "
                          <name_of_your_script>.sh/.py ${<esm_tools_variable_paths_your_script_needs>}"
```

If you have any question, please let us know.

**Comment by xshi-awi** [comment]
Thanks, I tried but it seems that the job was not submitted.


Here is the path:

/home/a/a270064/bb1029/AWIESM2.1-EXP/test/



Many thanks in advance.


Best

Xiaoxu

________________________________

**Comment by mandresm** [comment]
You need to include the `workflow` dictionary in one of the sections in the runscript (but not the `general` one). I recommend you create a new section for that as the exercise suggests, for example `flows`:
```
flows:
     workflow:
         [ ... ]
```

**Comment by xshi-awi** [comment]
Thanks, now it works.


Best

Xiaoxu

________________________________
